views/userManagementBadge.njk

```diff
@@ -71,25 +71,41 @@
                 <form id="deleteForm" action="/userManagement/deleteUser/{{user.badge}}" method="post" onsubmit="return confirm('This user will be deleted!')">
                 <button type='submit' class="btn btn-primary" name='submit' value="">Delete</button>
                 </form>
-                <br>
             {% if user.status == "User"%}
+                <br> 
                 <form id="ChangeStatusToManager" action="/userManagement/promoteUser/{{user.badge}}" method="post" onsubmit="return confirm('This user will be Manager!')">        
                 <button type='submit' class="btn btn-primary" name='submit' value="">Promote to Manager</button>
                 </form>
             {% elif user.status == "Manager" %}
+                <br>
                 <form id="ChangeStatusToManager" action="/userManagement/demoteManager/{{user.badge}}" method="post" onsubmit="return confirm('This user will be User!')">
                 <button type='submit' class="btn btn-primary" name='submit' value="">Demote to User</button>
                 </form>
             {% endif %}
+            
+            {% if user.loggedIn %}
+              <br>
+              <form id="BadgeOutUser" action="/userManagement/badgeOutUser/{{user.badge}}" method="post" >
+              <button type='submit' class="btn btn-primary" name='submit' value="">Badge Out</button>
+              </form>
+            {% endif %}
 
             </td>
         {% else %}
             <td>
                 <form id="validateForm" action="/userManagement/confirmUser/{{user.badge}}" method="post" onsubmit="return confirm('This user will be validated!')">
                 <button type='submit' class="btn btn-primary" name='submit' value="">Validate</button>
                 </form>
+            {% if user.loggedIn %}
```

Comment:
Why is this duplicated up top? Can you find a way to not duplicate this code?

Reply:
I did that so the badge out button can be fit into the table of the user, I tried to put it one time but unfortunately it did not fit in the table of the user information. I tested and it works fine

Reply:
I understand that, but could you think about how you can refactor the code to eliminate that duplication? You maybe just need to add another table row which has that button there.

Reply:
Another row would be good if that's possible, I haven't looked. Another option is to capture the parts that are different about the branches into a Nunjucks variable. Here's a (partial and untested) suggestion... not sure if its a good one or not - but here ya go:

The following factors out the commonalities only between the "User" and "Manager" branches of the "if user.confirmation" branch. Further refactoring is possible... but I thought this might help get you started. This allows us to collapse both if/else into a single block that renders the parts conditionally.

I'm assuming the "onsubmit="return confirm("...").." stuff is going away anyways [as they are replaced by req.flas() messages] so probably no need to factor that stuff. Anwyays:

```nunjucks
//...
//Here we capture the data
  {% if user.status == "User" %}
    {% set urlAction = "/userManagement/promoteUser/" + "{{ user.badge }}" %}
    {% set buttonText = "Promote to Manager" %}
    {% set formId = "ChangeStatusToManager" %}
  {% elif user.status == "Manager" %}
    {% set urlAction = "/userManagement/demoteManager/" + "{{ user.badge }}" %}
    {% set buttonText = "Demote to User" %}
    {% set formId = "DemoteManager" %}
  {% endif %}
<tr>
  <th></th>
    {% if user.confirmation  %}
  <td>
//...
//Here we have one form block statement instead of two
     <form id={{ formId }} action={{ urlAction }} method="post" onsubmit="return confirm('This user will be Manager!')">
     <button type='submit' class="btn btn-primary" name='submit' value="">{{ buttonText }}
     </button>
     </form>
   {% if user.loggedIn %}
   <br>
     <form id="BadgeOutUser" action="/userManagement/badgeOutUser/{{user.badge}}" method="post" >
     <button type='submit' class="btn btn-primary" name='submit' value="">Badge Out</button>
     </form>
   {% endif %}
  </td>
{% else %}
//...do more...
```